src/test/ui/pattern/usefulness/doc-hidden-fields.stderr

```diff
@@ -0,0 +1,44 @@
+error: pattern requires `..` due to inaccessible fields
+  --> $DIR/doc-hidden-fields.rs:8:9
+   |
+LL |     let HiddenStruct { one, two, } = HiddenStruct::default();
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+help: ignore the inaccessible and unused fields
+   |
+LL |     let HiddenStruct { one, two, .., } = HiddenStruct::default();
+   |                                ++++
```

Comment:
This error is incorrect (even if it were outside the crate of definition). `doc(hidden)` doesn't cause the field to be inaccessible; it's only meant to discourage public use by hiding the field from the generated documentation.

I feel mixed about `non_exhaustive_omitted_patterns` also relying on `doc(hidden)`; but at least in that case, it just suppresses the lint in some cases, rather than explicitly saying that the field is inaccessible.

Reply:
Do you mean the wording of messages with `doc(hidden)` fields, so it's  `pattern requires `..` due to `doc(hidden)` fields` or something like that?

I agree (I think you're kind of hinting at this) with you that making doc(hidden) behave in this way is odd/off. We have changed the meaning from (at least for error messages) this thing doesn't show its docs to, now, this is more like a `non_exhaustive` thing shrug.

Anyways I can conditionally alter the error messages for structs with doc hidden fields.

Reply:
> I agree (I think you're kind of hinting at this) with you that making doc(hidden) behave in this way is odd/off. We have changed the meaning from (at least for error messages) this thing doesn't show its docs to, now, this is more like a `non_exhaustive` thing shrug.

Yeah, that's what I mean. I think the compiler should really just ignore `doc(hidden)`; it's meant for rustdoc.

> Do you mean the wording of messages with `doc(hidden)` fields, so it's `` pattern requires `..` due to `doc(hidden)` fields`` or something like that?

That'd still be incorrect since the pattern doesn't actually _require_ `..`. `doc(hidden)` is just a nudge that some part of a library is an implementation detail. I don't think rustc should be changing its error messages based on `doc(hidden)`; it's surprising and not what `doc(hidden)` means.

Reply:
Would pinging this person https://github.com/rust-lang/rust/issues/89042 for some input be a good idea.

Reply:
Friendly ping, is there any place I should take this issue for more feedback I tried [zulip #t-lang/major changes](https://rust-lang.zulipchat.com/#narrow/stream/243200-t-lang.2Fmajor-changes/topic/Non.20exhaustive.20reachable.20patterns.20lint.20lang-team.23112/near/262076253) to no avail.

Reply:
You could try `#t-compiler` or `#t-lang` (rather than `#t-lang/major changes`). Sometimes though it's just hard to get feedback unfortunately.

> Would pinging this person #89042 for some input be a good idea.

You could try pinging m-ou-se, but she gets a lot of pings for library review, so not sure if she'd be able to respond.

You could also make this PR focus on just unstable fields (with the feature turned off) for now.

Reply:
Thanks for the quick response! I'll try the first few ideas and if it's still isn't moving I'll just focus on unstable fields.